designs/2019-linter-options/README.md

```diff
@@ -0,0 +1,126 @@
+- Start Date: 2019-05-12
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# The `linterOptions` property in Config Files
+
+## Summary
+
+This RFC adds `linterOptions` property to config files. People can use config files (including shareable configs!) instead of CLI options in order to configure some linter behavior.
```

Comment:
I'm not sure about the name `linterOptions`. Is this referring to options for the `Linter` API? If so, it seems like maybe `ignorePattern` doesn't fit.

For someone who doesn't know what the `Linter` API is, it might seem silly to use a name like `linterOptions` because it would seem like everything in a config file has "linter options" (since the config contains options for running ESLint).

Maybe it would be better to add these as top-level config properties instead.

Reply:
Yeah, I'm considering better names, but I don't have ideas for now.

I wanted to avoid adding to top-level. Because, especially, I don't want to put `excludedFiles` and `ignorePatterns` to the sibling. The `excludedFiles` affects the override entry, but the `ignorePatterns` affects the core behavior.

Reply:
How about `coreOptions`?

Reply:
I also think these properties make the most sense as top level items. I don't have a better idea at the moment, but I'm not sure `coreOptions` would make a lot of sense to end users (since it's not really clear what differentiates a "core option" from other top-level options).

Reply:
Hm.

Because my concern is confusion between `excludedFiles` and `ignorePatterns`, it may be fine that those options are on top-level and `ignorePatterns` cannot be in `overrides`.